src/iomusxml.cpp

```diff
@@ -591,6 +600,42 @@ bool MusicXmlInput::ReadMusicXml(pugi::xml_node root)
         measure->AddChild(iter->second);
     }
 
+    // manage endings stack
+    if (!m_endingStack.empty()) {
+        std::vector<std::pair<std::vector<Measure *>, musicxml::EndingInfo> >::iterator iter = m_endingStack.begin();
+        Object *section = iter->first.front()->GetParent();
+        for (; iter != m_endingStack.end(); ++iter) {
+            std::string logString = "";
+            logString = logString + "MusicXML import: Ending number='" + iter->second.m_endingNumber.c_str()
+                + "', type='" + iter->second.m_endingType.c_str() + "', text='" + iter->second.m_endingText + "' (";
+            std::vector<Measure *> measureList = iter->first;
+            std::vector<Measure *>::iterator jter = measureList.begin();
+            Ending *ending = new Ending();
+            if (iter->second.m_endingText.empty()) { // some musicXML exporters tend to ignore the <ending> text, so take @number instead.
+                ending->SetN(iter->second.m_endingNumber);
+            } else {
+                ending->SetN(iter->second.m_endingText);
+            }
+            ending->SetLendsym(LINESTARTENDSYMBOL_angledown); // default, does not need to be written
+            if (iter->second.m_endingType == "discontinue") {
+                ending->SetLendsym(LINESTARTENDSYMBOL_none); // no ending symbol
+            }
+            section->ReplaceChild(measureList.front(), ending);
+            ending->AddChild(measureList.front());
+            for (; jter != measureList.end(); ++jter) {
+                logString = logString + (*jter)->GetUuid().c_str();
+                if ((*jter)->GetUuid() != measureList.front()->GetUuid()) {
+                    int idx = section->GetChildIndex(*jter);
+                    section->DetachChild(idx);
+                    ending->AddChild(*jter);
+                }
+                logString = logString + ((*jter == measureList.back()) ? ")." : ", ");
+            }
+            LogMessage(logString.c_str());
+        }
+        m_slurStack.clear();
```

Comment:
I think you wanted to clear m_endingStack

Reply:
Many thanks, @lpugin, for looking over the code! 
Yes, my mistake. I will make all changes in a combined commit.